src/algorithm/rolling_window.go

```diff
@@ -0,0 +1,100 @@
+package algorithm
+
+import (
+	"math"
+
+	"github.com/coocood/freecache"
+	"github.com/envoyproxy/ratelimit/src/config"
+	"github.com/envoyproxy/ratelimit/src/utils"
+	"github.com/golang/protobuf/ptypes/duration"
+)
+
+var _ RatelimitAlgorithm = (*RollingWindowImpl)(nil)
+
+type RollingWindowImpl struct {
+	timeSource        utils.TimeSource
+	cacheKeyGenerator utils.CacheKeyGenerator
```

Comment:
same for `cacheKeyGenerator`, we no longer need this field in both window impls, as it has been moved to base window.